### PR TITLE
Http compression

### PR DESCRIPTION
**Pull request by gingerwizard**
Adds compression over http - currently using lz4 at block level only per https://clickhouse.com/docs/en/interfaces/http/#compression. Will add other compression options later - mainly zstd which ch-go supports as well as gzip etc for HTTP.

Note compression can be enabled via `OpenDb` with settings or via `compress=true` in the dsn.

We also introduce the `HttpsInterface` type and fix an issue with OpenDb not working @ortyomka i.e. it wasn't setting the scheme (which we no longer expose). I think needing to specify the interface in `OpenDb` is better than a scheme.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -110,6 +110,77 @@ This can be achieved by modifying the DSN to specify the http protocol.
...
+		Password: "",
+	},
+	Settings: clickhouse.Settings{
+		"max_execution_time": 60,
+	},
+	DialTimeout: 5 * time.Second,
+	Compression: &clickhouse.Compression{
+		Method: clickhouse.CompressionLZ4,
+	},
+	Interface: clickhouse.HttpInterface,
```
Comment (ernado) [inline README.md:130]: Please rename `HttpInterface` to `InterfaceHTTP`, in go we use all caps for abbrevations and is better to prefix the constant name with type, not postfix.

Also, can we use Protocol for this? I know that Interface is from clickhouse, but protocol is more common. Anyway, I'm ok with Interface.

Comment (gingerwizard) [reply README.md:130]: agreed protocol make sense. Noone is using this yet so its change now.

Comment (gingerwizard) [reply README.md:130]: @ernado moved to `Protocol` i.e. 
```
type Protocol int

const (
	Native Protocol = iota
	Http
	Https
)
```

File: clickhouse_options.go [hunk 2]
```diff
@@ -49,12 +49,12 @@ const (
...
 
-type InterfaceType int
+type Protocol int
 
 const (
-	NativeInterface InterfaceType = iota
-	HttpInterface
-	HttpsInterface
+	Native Protocol = iota
+	Http
```
Comment (ernado) [inline clickhouse_options.go:56]: Please use `HTTP` and `HTTPS` here as per https://github.com/golang/go/wiki/CodeReviewComments#initialisms

**Review by ernado — APPROVED** [review]

**Pull request merged**